sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StateTypesEncoderUtils.scala

```diff
@@ -49,12 +54,17 @@ object StateKeyValueRowSchema {
 class StateTypesEncoder[GK, V](
     keySerializer: Serializer[GK],
     valEncoder: Encoder[V],
-    stateName: String) {
-  import org.apache.spark.sql.execution.streaming.StateKeyValueRowSchema._
+    stateName: String,
+    hasTtl: Boolean) extends Logging {
+  import org.apache.spark.sql.execution.streaming.TransformWithStateKeyValueRowSchema._
 
   /** Variables reused for conversions between byte array and UnsafeRow */
   private val keyProjection = UnsafeProjection.create(KEY_ROW_SCHEMA)
-  private val valueProjection = UnsafeProjection.create(VALUE_ROW_SCHEMA)
+  private val valueProjection = if (hasTtl) {
+      UnsafeProjection.create(VALUE_ROW_SCHEMA_WITH_TTL)
```

Comment:
nit: indent seems off ?

Reply:
I indented it with 2 spaces as it was easier to read that way. I did not find any specific guidance for class variable declarations with `if .. else` in https://github.com/databricks/scala-style-guide?tab=readme-ov-file#indent.

Reply:
Hmm - maybe its just a GH UI thing - but it appears like 4 spaces to me ?

Reply:
I think its because `UnsafeProjection.create` is inside the if part. The curly brace is next line is indented with 2 spaces.